stelaval/assignment1/Anagram.java

```diff
@@ -0,0 +1,95 @@
+package assignment1;
+
+import java.util.Arrays;
+
+public class Anagram {
+
+	/**
+	 * Determines if one string is an anagram of another string
+	 * 
+	 * @param s1 first string
+	 * 
+	 * @param s2 second string
+	 * 
+	 * @return true if s1 and s2 are anagrams of each other
+	 */
+	static boolean isAnagram(String s1, String s2) {
+
+		String string1 = s1.replaceAll("\\s", "");
+		String string2 = s2.replaceAll("\\s", "");
+
+		if (string1.length() != string2.length()) {
+			return false;
+		} else {
+
+			char[] s1Array = string1.toCharArray();
+			char[] s2Array = string2.toCharArray();
+
+			Arrays.sort(s1Array);
+			Arrays.sort(s2Array);
+
+			return Arrays.equals(s1Array, s2Array);
+		}
+	}
+
+	/**
+	 * Handles both case sensitive and case insensitive anagrams
+	 */
+	
+	static boolean isWordAnagram(String s1, String s2, boolean isCaseSensitive) {
+		if (isCaseSensitive) {
+			return isAnagram(s1, s2);
+		} else {
+			return isAnagram(s1.toLowerCase(), s2.toLowerCase());
+		}
+	}
+
+	/**
+	 * Determines if one sentence is an anagram of another sentence

```

Comment:
It would be good to add more details, so that if someone decides to use your code in their project, they immediately know what does it mean that two sentences are anagrams of each other.

Reply:
Will have that in mind